library/alloc/src/vec/source_iter_marker.rs

```diff
@@ -106,3 +94,52 @@ fn write_in_place_with_drop<T>(
         Ok(sink)
     }
 }
+
+/// Helper trait to hold specialized implementations of the in-place iterate-collect loop
+trait SpecInplace<T, I>: Iterator<Item = T> {
```

Comment:
Shouldn't this be `SpecInPlace`? Also, I find the name a bit confused, I wonder why isn't it `CollectInPlace`?

Reply:
> Shouldn't this be SpecInPlace? 

Maybe? Depends on whether you consider in-place a single term or two words.

> I wonder why isn't it CollectInPlace?

Internal specialization traits often have `Spec` somewhere in their name to indicate the purpose. I guess `SpecInplaceCollect` could work too.